File: 242.有效的字母异位符/242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
#python版哈希表解法，C++ 的 unordered_map 等价于 Python 的 dict
#C++ 的 unordered_set 等价于 Python 的 set
        if len(s)!=len(t)  return False
        char_map = {}
        for c in s:
            char_map[c] = char_map.get(c,0) + 1
#get(key,default),获取键的值，若键不存在则返回默认值
        for c in t:
            char_map[c] = char_map.get(c,0) - 1
            if char_map[c] < 0
                return False
    return True
#-----------------------最优----------------
        #return Counter(s)==Counter(t)
        #Python 的 Counter 是 collections 模块中的一个类，用于统计可迭代对象中元素的出现次数

File: 242.有效的字母异位符/242.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.length() != t.length())
            return false;
        // 优先处理边界情况，就是两个字符串长度不相等时
        unordered_map<char, int> charMap;
        for (char c : s)
        {
            charMap[c]++; //[]在哈希表使用时，会创建键值对，记录每个元素及对应的出现次数，值初始为0，++遇到一次就会加1，
        }
        for (char c : t)
        {
            charMap[c]--; //--则是减1
        }
        for (auto &[words, count] : charMap)
        {
            if (count != 0)
            {
                return false;
            }
        }
        return true;
    }
};